cardinal/ecs/query.go

```diff
@@ -217,3 +197,32 @@ func (r *QueryType[Request, Reply]) EncodeAsABI(input any) ([]byte, error) {
 	}
 	return bz, nil
 }
+
+func validateQuery[Request any, Reply any](name string, handler func(wCtx WorldContext, req *Request) (*Reply, error)) error {
+	if name == "" {
+		return errors.New("cannot create query without name")
+	}
+	if handler == nil {
+		return errors.New("cannot create query without handler")
+	}
+
+	var req Request
+	var rep Reply
+	reqType := reflect.TypeOf(req)
+	reqKind := reqType.Kind()
+	reqValid := false
+	if (reqKind == reflect.Pointer && reqType.Elem().Kind() == reflect.Struct) || reqKind == reflect.Struct {
+		reqValid = true
+	}
+	repType := reflect.TypeOf(rep)
+	repKind := reqType.Kind()
+	repValid := false
+	if (repKind == reflect.Pointer && repType.Elem().Kind() == reflect.Struct) || repKind == reflect.Struct {
+		repValid = true
+	}
+
+	if !repValid || !reqValid {
+		return errors.New(fmt.Sprintf("Invalid Query: %s: The Request and Reply must be both structs", name))
```

Comment:
go errors should not start with a capital:
https://google.github.io/styleguide/go/decisions.html#:~:text=Error%20strings%20should%20not%20be,being%20printed%20to%20the%20user.